### ISSUE TITLE
Link account with Starter Edition

### ISSUE THREAD

**Issue by bchevreau** [stevin05/CraftScan#37]
I was wondering if WoW Starting Edition account-linking is supposed to work?
I've been wanting to try to link my main account with a starter edition account that has a single character (lvl 5 with a classic profession just to be sure) so that I could take orders on the starter edition account while relogging to each character on my main to craft, for better communication.

However the list of crafters, my custom greetings or anything never shows up on the linked starter edition account.

Sometimes the linked account appears as a name, albeit always offline. But on the starter edition account, my main account never does.

I do receive an invitation if I link from main to starter, but never from starter to main

![image](https://github.com/user-attachments/assets/b3a08e00-d7b6-4db3-935c-1f91e4e53945)

Does the starter need a subscription active too maybe?

My goal was to test before actually getting a sub on that account too but now I'm perplexed.

PS: do you have a patreon or Paypal for donations? You definitely deserve more than a coffee for this AMAZING addon! <3

**Comment by stevin05** [comment]
The starter account does not require a subscription, but it can't message people, so you'll have to do all the messaging from your main account after the start proxies an order over.

When setting it up, you had 'Full Control' checked? Right click on the account and make sure the characters you are playing are on the list known to each side. By default, it'll grab that characters you initially linked and any crafters CraftScan knows about, but you can manually add others. Can try right clicking it and deleting the link on both sides and re-creating it. Turning it off and on again always fixes things.

I haven't tested it in a while, so maybe I broke something with it. Not sure I ever tried initiating the link on the starter account.

**Comment by stevin05** [comment]
Just gave it a try linking from a trial account to subscribed account and it worked.

You can try installing DevTool (it's on CurseForge) and then edit the top of CraftScan/Utils/Debug.lua to set 'local DEBUG = true' instead of false. Link the accounts, and then type /dev on both sides and you can hopefully see the messages flowing back and forth.

Here's what it looks like successfully working. The 'Sending msg' and 'Received msg' lines show what data is flowing back and forth. It starts by sending revision numbers to determine what is out of date before sending the full data of anything out of date.

Will be playing in an hour or 2 if you want to hop on discord to troubleshoot.

![image](https://github.com/user-attachments/assets/9825ca94-87e8-42e7-9136-31ef0ead6cf3)

![image](https://github.com/user-attachments/assets/ec2bffa2-1162-4d00-b3b5-54b3a190d4b0)

**Comment by bchevreau** [comment]
> Just gave it a try linking from a trial account to subscribed account and it worked.
> 
> You can try installing DevTool (it's on CurseForge) and then edit the top of CraftScan/Utils/Debug.lua to set 'local DEBUG = true' instead of false. Link the accounts, and then type /dev on both sides and you can hopefully see the messages flowing back and forth.

Will try with DevTool as soon as I get a bit of time for it, ty for the pointers. Thanks a lot for taking the time! 

I did try to unlink and redo the process from start>main or from main>start but it never populates the start account with the name of crafters from my main for some reason.
The main account always shows start account as offline:
![image](https://github.com/user-attachments/assets/b30b5dd0-a609-4e55-9481-541df47d32a8)

And the start account also never shows the linked main account up:
![image](https://github.com/user-attachments/assets/130828a6-28a8-46e0-a418-f255f24bc207)

**Comment by stevin05** [comment]
Maybe the server name. I did test on connected realms, but from reading about realm names while setting it up, seems like there are lots of special cases around them.

One thing to try:
1. Log out on the starter account.
2, Toggle the "All Crafters" check box a few times to enable/disable scanning on the main account. This will increment the revision number of all the professions.
3. Log back into the starter account.

I saw it a few times during testing where revision numbers would become equal on both sides and then sync's would never happen. I thought I'd fixed it, but the toggling should bump up the revision numbers if that is what has happened.
